Libplanet.Tests/Blocks/BlockHeaderTest.cs

```diff
@@ -0,0 +1,147 @@
+using System;
+using System.Collections.Immutable;
+using System.Globalization;
+using Libplanet.Blocks;
+using Xunit;
+
+namespace Libplanet.Tests.Blocks
+{
+    public class BlockHeaderTest : IClassFixture<BlockFixture>
```

Comment:
Haven't we had such `Validate()` tests for `Block` before?

Reply:
Yes, and we still have tests for `Block<T>.Validate()`.

I've added these tests to have confidence for separated codes while working because it's more direct than `Block<T>.Validate()`'s tests.